src/fileManager.ts

```diff
@@ -1,103 +1,270 @@
+  private async initOwnerFeedList(): Promise<void> {
+    if (this.nodeAddresses === undefined) {
+      throw new SignerError('Node addresses not found');
+    }
 
-      this.fileInfoList.push(fileInfo);
-      console.log(this.fileInfoList[0].batchId);
-      const fileInfoList = this.fileInfoList.map((item) => ({
-        batchId: item.batchId.toString(),
-        file: {
-          reference: item.file.reference.toString(),
-          historyRef: item.file.historyRef.toString(),
-        },
-        topic: item.topic?.toString(),
-        owner: item.owner?.toString(),
-        name: item.name,
-        timestamp: item.timestamp,
-        shared: item.shared,
-        preview: item.preview,
-        redundancyLevel: item.redundancyLevel,
-        customMetadata: item.customMetadata,
-      }));
-
-      localStorage.setItem(FILE_INFO_LOCAL_STORAGE, JSON.stringify(fileInfoList));
-
-      return index;
-    } catch (error) {
-      console.error('Error saving file info:', error);
-      throw error;
+    const latestFeedData = await this.getFeedData(this.ownerFeedTopic);
+    const dataArr = latestFeedData.payload.toUint8Array();
+
+    if (SWARM_ZERO_ADDRESS.equals(dataArr)) {
+      console.log("Owner fileInfo feed list doesn't exist yet.");
+      return;
+    }
+
+    this.ownerFeedNextIndex = latestFeedData.feedIndexNext?.toBigInt() || 0n;
+    const refWithHistory = latestFeedData.payload.toJSON() as ReferenceWithHistory;
+
+    const fileInfoFeedListRawData = await this.bee.downloadData(new Reference(refWithHistory.reference), {
+      actHistoryAddress: new Reference(refWithHistory.historyRef),
+      actPublisher: this.nodeAddresses.publicKey,
+    });
+    const fileInfoFeedListData = fileInfoFeedListRawData.toJSON() as WrappedFileInfoFeed[];
+
+    for (const feedItem of fileInfoFeedListData) {
+      try {
+        assertWrappedFileInoFeed(feedItem);
+        this.ownerFeedList.push(feedItem);
+      } catch (error: any) {
+        console.error(`Invalid WrappedFileInoFeed item, skipping it: ${error}`);
+      }
+    }
+
+    console.log('FileInfo feed list fetched successfully.');
+  }
+
+  // fetches the file info list from the owner feed and unwraps the data encrypted with ACT
+  private async initFileInfoList(): Promise<void> {
+    if (this.nodeAddresses === undefined) {
+      throw new SignerError('Node addresses not found');
+    }
+
+    for (const feedItem of this.ownerFeedList) {
+      const rawFeedData = await this.getFeedData(new Topic(feedItem.topic));
+      const fileInfoFeedData = rawFeedData.payload.toJSON() as ReferenceWithHistory;
+
+      const unwrappedFileInfoData = (
+        await this.bee.downloadData(fileInfoFeedData.reference, {
+          actHistoryAddress: new Reference(fileInfoFeedData.historyRef),
+          actPublisher: this.nodeAddresses.publicKey,
+        })
+      ).toJSON() as ReferenceWithHistory;
+
+      try {
+        assertFileInfo(unwrappedFileInfoData);
```

Comment:
This line throwing an error in the tests: `preview property of FileInfo has to be string!`, but it is a `ReferenceWithHistory` type.